petastorm/spark/spark_dataset_converter.py

```diff
@@ -69,19 +70,32 @@ def _cache_df_or_retrieve_cache_path(df, cache_dir, row_group_size):
     """
     uuid_str = str(uuid.uuid4())
     save_to_dir = os.path.join(cache_dir, uuid_str)
-    df.write.mode("overwrite") \
+
+    old_compression = _spark_session.conf.get("spark.sql.parquet.compression.codec")
+
+    _spark_session.conf.set(
+        "spark.sql.parquet.compression.codec", compression)
+    df.write \
         .option("parquet.block.size", row_group_size) \
         .parquet(save_to_dir)
     atexit.register(shutil.rmtree, save_to_dir, True)
 
+    # Restore compression setting.
+    _spark_session.conf.set(
```

Comment:
Is there any chance that `df.write` fails and thus `compression.codec` is not restored successfully? Maybe using a try-catch?